lua/starfall/libs_cl/notification.lua

```diff
@@ -0,0 +1,80 @@
+-------------------------------------------------------------------------------
+-- Notification functions
+-------------------------------------------------------------------------------
+
+-- Register Priveleges
+SF.Permissions.registerPrivilege("notification", "Create notifications", "Allows the user to create notifications on their screen", { client = { default = 1 } })
+
+local checktype = SF.CheckType
+local checkluatype = SF.CheckLuaType
+local checkpermission = SF.Permissions.check
+
+--- Notification library. Allows the user to display hints on the bottom right of their screen
+-- @client
+local notification_library = SF.RegisterLibrary("notification")
+
+SF.AddHook("postload", function()
+	-- @name SF.DefaultEnvironment.NOTIFY
+	-- @class table
+	SF.DefaultEnvironment.NOTIFY = {
+		["GENERIC"] = NOTIFY_GENERIC,
+		["ERROR"] = NOTIFY_ERROR,
+		["UNDO"] = NOTIFY_UNDO,
+		["HINT"] = NOTIFY_HINT,
+		["CLEANUP"] = NOTIFY_CLEANUP
+	}
+end)
+
+--- Displays a standard notification.
+-- @param text The text to display
+-- @param type Determines the notification method.
+---NOTIFY.GENERIC
+---NOTIFY.ERROR
+---NOTIFY.UNDO
+---NOTIFY.HINT
+---NOTIFY.CLEANUP
+-- @param length Time in seconds to display the notification
+function notification_library.addLegacy(text, type, length)
```

Comment:
There should be some length limit, I think 30 seconds is enough. Make sure the length is not a NaN value.